### PR TITLE
OAuth Providers Github App email problem

### PR DESCRIPTION
**Pull request by monoald**
When trying authentication with Github App, Github only sends the user's profile public email if there is any we receive `null`.
I am now explicitly requesting user emails to get a non-null value.

### CODE REVIEW

File: packages/oauth-providers/src/providers/github/authFlow.ts [hunk 1]
```diff
@@ -103,6 +109,26 @@ export class AuthFlow {
       throw new HTTPException(400, { message: response.message })
     }
 
+    if (!this.oauthApp) {
+      const emails = (await fetch('https://api.github.com/user/emails', {
+        headers: {
+          Authorization: `Bearer ${this.token?.token}`,
+          'User-Agent': 'Hono-Auth-App',
```
Comment (yusukebe) [inline packages/oauth-providers/src/providers/github/authFlow.ts:116]: It's better to make `Hono-Auth-App` as a constant.

File: .changeset/sharp-fans-explain.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@hono/oauth-providers': minor
```
Comment (yusukebe) [inline .changeset/sharp-fans-explain.md:2]: I think it will be `patch` release because this PR is `fix`.

How do you think?

Comment (monoald) [reply .changeset/sharp-fans-explain.md:2]: Right! My bad. I'll change it now

**Comment by monoald** [comment]
Changes made it, `"Hono-Auth-App"` as a constant  and minor bump changed for patch bump.

**Review by yusukebe — COMMENTED** [review]
LGTM!

**Comment by yusukebe** [comment]
@monoald 

Thanks! Let's go with it.

**Pull request merged**